### PR TITLE
Add 'companies' collection

### PR DESCRIPTION
**Pull request by maktwin**

### CODE REVIEW

File: lib/collections/companies.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  config.declaration.extended((declaration) => declaration
+    .parameters({
+      recentlyCreated: {
+        description: 'Fetch companies which were created in the last 30 days, or the 10k most recently created records sorted by the date the companies were created.'
+      },
+      recentlyUpdated: {
+        description: 'Fetch companies which were modified in the last 30 days, or the 10k most recently modified records sorted by the date the companies were most recently modified.'
+      }
+    })
+    .conflicts('recentlyCreated', 'recentlyCreated')
```
Comment (alexey-pelykh) [inline lib/collections/companies.js]: Have you tested this?

File: lib/collections/companies.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    .parameters({
+      properties: {
+        description: 'Fetch companies with specific properties',
+        type: 'string'
+      },
+      propertiesWithHistory: {
```
Comment (alexey-pelykh) [inline lib/collections/companies.js]: How will CLI react if I'll call without parameters at all?
How to retrieve all properties with history?

File: lib/collections/companies.js (hunk 1 above)
Comment (maktwin) [reply lib/collections/companies.js]: I deleted this line

File: lib/collections/companies.js (hunk 2 above)
Comment (maktwin) [reply lib/collections/companies.js]: If you will call without parameters then you will fetch all companies. By default, each company object without properties. We need to specify which properties we want to retrieve.

Comment (alexey-pelykh) [reply lib/collections/companies.js]: That's not the desired behavior - it's not similar to contacts. By default, _all_ properties should be provided with history

**Review by alexey-pelykh — APPROVED** [review]

**Pull request merged**